kork/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/artifactstore/ArtifactStore.java

```diff
@@ -28,20 +32,44 @@ public class ArtifactStore implements ArtifactStoreGetter, ArtifactStoreStorer {
 
   private final ArtifactStoreStorer artifactStoreStorer;
 
+  private final Map<String, List<ApplicationStorageFilter>> exclude;
+
   public ArtifactStore(
-      ArtifactStoreGetter artifactStoreGetter, ArtifactStoreStorer artifactStoreStorer) {
+      ArtifactStoreGetter artifactStoreGetter,
+      ArtifactStoreStorer artifactStoreStorer,
+      Map<String, List<ApplicationStorageFilter>> exclude) {
     this.artifactStoreGetter = artifactStoreGetter;
     this.artifactStoreStorer = artifactStoreStorer;
+    this.exclude = exclude;
+  }
+
+  public boolean shouldExclude(String type, String application) {
+    return application == null
+        || this.exclude.containsKey(type)
+            && this.exclude.get(type).stream().anyMatch((filter) -> filter.filter(application));
   }
 
   /** Store an artifact in the artifact store */
-  public Artifact store(Artifact artifact) {
-    return artifactStoreStorer.store(artifact);
+  public Artifact store(Artifact artifact, ArtifactDecorator... decorators) {
+    String application = AuthenticatedRequest.getSpinnakerApplication().orElse(null);
+    if (application == null) {
+      log.warn("failed to retrieve application from request artifact={}", artifact.getName());
+      return artifact;
```

Comment:
Please update the javadoc to describe the return value, and what happens in these "failure" cases.